### PR TITLE
Ports IPCs from Beestation

### PR DESCRIPTION
**Pull request by redmoogle**
### Intent of your Pull Request

Credits to BeeStation(Credits would be a milelong if I listed each individual port)

Adds IPCs
Vid: https://www.youtube.com/watch?v=eWWDqZTMs3k

They're a synthetic race **different** from preternis, mainly with chassis customization and TV screens. They have a high burn mod

also they cant be cloned because they're robots. so you can use a cyborg rebooter in place of that.

They lose limbs easily but can be attached easily, Head is stored in the _chest_ and _can_ live without a head

Wiki Documentation:

There a race. They are robotic (they already have been given pictures needed, and from what I've heard are already working on the page)

System Cleaner | 1 part Ethanol, 1 part Chlorine, 2 parts Phenol, 1 part Potassium | Quickly purges toxin damage in the rare case an IPC gets said damge.

Solder: | 1 part Ethanol, 1 part Copper, 1 part Silver, Temperature 374K | Quickly heals brain damage and traumas, but only for IPCs

- [x] Remove that liver runtime
- [x] Test Mechfabs
- [x] Test Chemicals
- [x] Test Charging
- [x] Test ORGANIC/SYNTHETIC Chemicals
- [x] Fix Limb Issue
- [x] Test Revival/Surgery     

### Why is this good for the game?

Is a very cool race and could easily have some cool interactions

#### Changelog

:cl:  
rscadd: Add IPCs from Beestation
/:cl:

### CODE REVIEW

**Comment by Moja10** [comment]
Hopefully Preternis and IPCs can coexist...

**Comment by redmoogle** [comment]
> Hopefully Preternis and IPCs can coexist...

I mean I might give preternis some more uniqueness to differentiate them

**Comment by ynot01** [comment]
yo they look ugly asf

**Comment by redmoogle** [comment]
> yo they look ugly asf 

Didnt really put effort into the picture. Will get to that tommorow

**Review by jcatjmeow — CHANGES_REQUESTED** [review]
1) they look ugly af

2) they are horribly unbalanced. Being able to heal from death is HUGELY a big deal, especially since it makes them basically immune to CMD.

**Comment by ghost** [comment]
They're also incredibly fragile, Jcat, and dealing with them permanently is as easy as an extra hit to the chest.

If it's such an issue, just give IPCs special CMD rules.

**Comment by UselessTheremin** [comment]
i mean these are cool and all, but why do we need them other than "sure why not, another cool race?" i honestly dont know what positive contributions they will really bring to this server other than another race option to choose.

**Comment by redmoogle** [comment]
> i mean these are cool and all, but why do we need them other than "sure why not, another cool race?" i honestly dont know what positive contributions they will really bring to this server other than another race option to choose. 

Whats the point of any race.

**Comment by ghost** [comment]
they'll contribute more than the fucking xenomorph race and snail race that was added for no reason that's for sure

or the fact that we have both podpeople and phytosians

**Comment by Tipy1802** [comment]
> i mean these are cool and all, but why do we need them other than "sure why not, another cool race?" i honestly dont know what positive contributions they will really bring to this server other than another race option to choose.

More content, more ways to play the game, more things to do, more fun

**Comment by TheLinuxOS** [comment]
I disagree with the assessment they look ugly asf, and I feel that is more of an opinion than an issue that needs to be fixed. We have xenos and moths in the game and they both look ugly as hell in MY opinion.

**Comment by Moggram** [comment]
Oh yes, please do add IPC's. This way everyone will stop crying about Preternis when they find out how broken this race is.




@redmoogle 
> 
> I mean I might give preternis some more uniqueness to differentiate them

No need for that. We are already so different as it is.

**Comment by redmoogle** [comment]
> 
> 
> Oh yes, please do add IPC's. This way everyone will stop crying about Preternis when they find out how broken this race is.
> 
> @redmoogle
> 
> > I mean I might give preternis some more uniqueness to differentiate them
> 
> No need for that. We are already so different as it is.

Good thing thats been completely addressed

**Comment by Moggram** [comment]
@redmoogle 

Them being broken or the dissimilarity of ICP's and Preternis?

**Comment by redmoogle** [comment]
> 
> 
> @redmoogle
> 
> Them being broken or the dissimilarity of ICP's and Preternis?

Both.

They die really quickly to an emp, a heavy one dealing 75 brain damage, 15 toxin, and causing other shitty effects

They take massive burn/heat damage, so lasers fuck them over

Self Revive? You can take their brain out in less than 30 seconds with makeshift tools

As for uniqueness, Their heads are changeable, they have unique limb reskins, Their brain is in the chest and is a posibrain.

So please STFU and get out :) because you cant read

**Comment by Moggram** [comment]
@redmoogle 
Oh, the EMP is that bad. If you don't want to get knockdown, just remove your ears. You can't hear any more, but EMP is not death and you can still run away or dodge.

The 2x burn is a big draw back, I do give you that one. I cry when lasguns hit me and its only 1.5x
But some what makes up for it whit a toxin mod of 0 and bunt being 1.

I do have some questions about the chems you are adding.
Will system cleaner and liquid solder work on Preternis? 
Also I see that omnizine got the SYNTHETIC flag to, so maybe it will heal them?

**Comment by redmoogle** [comment]
> 
> 
> @redmoogle
> Oh, the EMP is that bad. If you don't want to get knockdown, just remove your ears. You can't hear any more, but EMP is not death and you can still run away or dodge.
> 
> The 2x burn is a big draw back, I do give you that one. I cry when lasguns hit me and its only 1.5x
> But some what makes up for it whit a toxin mod of 0 and bunt being 1.
> 
> I do have some questions about the chems you are adding.
> Will system cleaner and liquid solder work on Preternis?
> Also I see that omnizine got the SYNTHETIC flag to, so maybe it will heal them?

Well since IPCs are well, kinda all robotic of course they wont take toxin damage. They're basically cyborgs but lacking all the utility, I will be making them take slightly more brute 1.25-1.5 for being a little out there

As for system cleaner and liquid solder. as it stands in their lore and name it make sense it would really affect preternis. Maybe make it weaker for them?
And yes ozimine heals IPCs. Although I will have to flesh out more robotic chems for preternis

~~Also sorry for going nuclear on you, you kinda made me a _little_ mad~~

**Comment by Moggram** [comment]
>
>Also sorry for going nuclear on you, you kinda made me a little mad

Hey, its cool. That's just normal internet talk to me. At lest I didn't just call them ugly and call it a day. (I don't think they are by the way.)

>
>Well since IPCs are well, kinda all robotic of course they wont take toxin damage. They're basically cyborgs but lacking all the
>utility, I will be making them take slightly more brute 1.25-1.5 for being a little out there

After thinking about it, I don't think a 1.25  bunt mod is needed. lore wise, (for bee and baystation) they are made alot better then Preternis and it dose not fit them. How about they can get ion laws like the AI and need to me be fix by reboot card from R&D? Its just an ideal.

**Review by jcatjmeow — APPROVED** [review]
I fucking hate IPCS

**Comment by ghost** [comment]
I fucking hate IPCS

**Comment by ghost** [comment]
@Moggram You're a dent.

IPCs are well balanced.

They trade off their easy revivability with ridiculous drawbacks and incredible fragility. They'll get bowled over by a single laser gun.

EMPs are also incredibly common. Every antagonist in the game bar shadowling, nightmare, morph, blood brothers, clock cult and rev can field one by default, and the rest can easily get access to it.

**Comment by Yogbot-13** [comment]
#### Test Merge Deployed

##### Server Instance
main

##### Revision
Origin: ef0db3a25ba81dee1c39ea8fc8162a7a4a752ce0
Pull Request: 4cce3216e06c35a14a2ddcfafbc0f01b912b3702
Server: aaf2a4de3a62a53112d314c3f9281454f06ce656

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 61be296c7775d9ea88e9b8a729456b211e4cc029
Pull Request: 4cce3216e06c35a14a2ddcfafbc0f01b912b3702
Server: 7556b2e6eebbdd8cd977546ea7280d0bb63bd23f

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 61be296c7775d9ea88e9b8a729456b211e4cc029
Pull Request: 26479838808cc46cd1173ec300e83992c50657f8
Server: 30cee7883419fedc2b14d05329f2362084d4a522

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 846b6d0591d4fdfa09eb9564a32ffc4691283565
Pull Request: 26479838808cc46cd1173ec300e83992c50657f8
Server: a3380909465cd8b2082b69723d93038188c356e5

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 846b6d0591d4fdfa09eb9564a32ffc4691283565
Pull Request: 9f9a48c67696edc8e6384718fd82043825b96401
Server: 952eb07ef4c61276755bc1d018d27c948afb0e36

**Comment by Mqiib** [comment]
I noticed you still have the old polysmorph body parts in your human_parts.dmi. Might want to get rid of those, they aren't used any more.

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 846b6d0591d4fdfa09eb9564a32ffc4691283565
Pull Request: fbc171afdc22b1ce6cdb43eb535408f613980371
Server: 9a2ec9b6425c6d90690405f9c3b94f4666402c94

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: dd1d7fc33764ab1f7a35cf63a176bb8ea99c5bb7
Pull Request: fbc171afdc22b1ce6cdb43eb535408f613980371
Server: 14cdaf78532bb345dde5558d0304548c1bc44f79

**Comment by Mqiib** [comment]
*dab

**Comment by Cartlord** [comment]
Yo, noticed one feature missing - Bee's IPCs can recharge from cyborg stations (such as the ones in Robotics, or the one in Restrooms) as well as APCs , while these can only use APCs.

**Comment by redmoogle** [comment]
> Yo, noticed one feature missing - Bee's IPCs can recharge from cyborg stations (such as the ones in Robotics, or the one in Restrooms) as well as APCs , while these can only use APCs.

Probably missed it in one of 20 prs i ported

**Comment by Cartlord** [comment]
> 
> 
> > Yo, noticed one feature missing - Bee's IPCs can recharge from cyborg stations (such as the ones in Robotics, or the one in Restrooms) as well as APCs , while these can only use APCs.
> 
> Probably missed it in one of 20 prs i ported

understandable have a nice day
TBH it might be better that that isn't here, because on Bee it's able to sometimes make an IPC fat, and honestly playing a fat IPC is awful.

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: dd1d7fc33764ab1f7a35cf63a176bb8ea99c5bb7
Pull Request: 03c6e9d4369aee490fef2e3d675f5d4ed3a38e2a
Server: 38094ac42f2645436ff266e4acf28dbe0a1101bc

**Comment by Reed0506** [comment]
if you add a slight movespeed penalty like phytosian has, people will have even less to complain about in terms of balance, though thats probably not needed.

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: a6e2415c469ba1af6e29b54aa239ed65b8c36ee7
Pull Request: 03c6e9d4369aee490fef2e3d675f5d4ed3a38e2a
Server: d24b215e0e8c6c5423888ae49e4f29a9cf171b3c

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 66b7150aeab900c3f31c14b1bd1e42b0bf923c18
Pull Request: 03c6e9d4369aee490fef2e3d675f5d4ed3a38e2a
Server: ef2e93fba57b1a0477b759c6d1a9a04f2aa0c565

**Comment by Cartlord** [comment]
Unsure if this is a glitch or a feature, but I got ear damage (I think from an EMP), and even after a roboticist gave me a new pair of Auditory Sensors, the damage persisted.

EDIT: As the round went on, the ears healed on their own, for no discernible reason.

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 158c00ddf79f3de3a5da5964b55ac17a89978863
Pull Request: 03c6e9d4369aee490fef2e3d675f5d4ed3a38e2a
Server: b8c32c664435221e3bc7144827e3654e90e4e6dc

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: ca59602e0478d0927aad7a9b1fd4711bde1f9c82
Pull Request: 03c6e9d4369aee490fef2e3d675f5d4ed3a38e2a
Server: c2377bfb63991a30acd46b8f81123ed6b04133b5

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 39c610f599c05ab55c9551c8328eb9cd5b838117
Pull Request: 03c6e9d4369aee490fef2e3d675f5d4ed3a38e2a
Server: 576b980d7860d3a2d8af75956984af14a64580ef

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 39c610f599c05ab55c9551c8328eb9cd5b838117
Pull Request: 03c6e9d4369aee490fef2e3d675f5d4ed3a38e2a
Server: ce13dee3d6e20b89dc7daedfb859ee58550b7dcf

File: code/modules/mob/living/taste.dm [hunk 1]
```diff
@@ -25,7 +27,7 @@
 			text_output = pick("spiders","dreams","nightmares","the future","the past","victory",\
 			"defeat","pain","bliss","revenge","poison","time","space","death","life","truth","lies","justice","memory",\
 			"regrets","your soul","suffering","music","noise","blood","hunger","the american way")
-		if(text_output != last_taste_text || last_taste_time + 100 < world.time)
+		if((text_output != last_taste_text || last_taste_time + 100 < world.time) && !NO_TASTE_SENSITIVITY)
```
Comment (SomeguyManperson) [inline code/modules/mob/living/taste.dm:30]: this seems to stop anything from tasting stuff

**Comment by SomeguyManperson** [comment]
also saw an IPC changeling and I'm pretty sure IPCs being selected as changeling objectives

**Comment by swissloaf** [comment]
Why!!!!!!!!

**Comment by LazennG** [comment]
WHY NO MORE TVHEAD

**Comment by ChesterTheCheesy** [comment]
brrrrrr

**Pull request closed without merging**